module-2/backend/week_two.md

```diff
@@ -26,29 +35,93 @@ end
 
 Now how would you find the owner of the team with an id of 4?
 
-3. What do they allow you to do?
-7. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
-8. Define foreign key, primary key, and schema.
-9. Describe the relationship between a foreign key on one table and a primary key on another table.
-10. What are the parts of an HTTP response?
-11. Describe some techniques to make our Sinatra code more DRY. Give an example of when you would use these techniques.
+Team.find(4).owner.name
+
+
+1. What do they allow you to do? 
+I don’t know what this question is asking.
+
+2. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+There’s a many-to-many relationship, as students have many teachers and teachers have many students through classes.
+
+class Students: has_many :teachers
+has_many :classes- through teachers
+
+class Teachers
+has_many :students
+has_many classes through students
+
+students			classes				  teachers		
+--id				--id					    --id
+--name			--name				    --name
+				    --teacher_id
+				    --student_id
+
+
+3. Define foreign key, primary key, and schema.
+A foreign key is the key associated with the direct relationship between two tables/sets of data. The primary key is the id # of the table/data. A schema is a mapping of the tables and how to organize them and the primary keys and foreign keys’ locations.
+
+4. Describe the relationship between a foreign key on one table and a primary key on another table.
+If there’s a one-to-many relationship, the primary key is on the one side of the association while the foreign key is on the many side. So say if a teacher has many students but a student has one teacher. This is a one-to-many relationship. The foreign key will be under the student table as teacher_id.
+
+
+5. What are the parts of an HTTP response?
+There are three main components, the request headers, the request body, and request method. The request method contains the method/verb, the headers are used to communicate information about the client environment, and the body contains the actual request being sent to the HTTP server.
```

Comment:
This looks good for a request, but we were asking about a *response*. A response will also have headers and a body, but the status code (e.g. 200, 404, 503, etc.) will be the first header.